rule.php

```diff
@@ -62,6 +65,39 @@ public static function make (quiz $quizobj, $timenow, $canignoretimelimits) {
      * @param MoodleQuickForm $mform the wrapped MoodleQuickForm.
      */
     public static function add_settings_form_fields(mod_quiz_mod_form $quizform, MoodleQuickForm $mform) {
+        global $DB, $USER;
+        $settingsmanager = new settings_manager();
+        $elements = $settingsmanager->get_quiz_element_types();
+        $defaults = $settingsmanager->get_quiz_defaults();
+        $hideifs = $settingsmanager->get_quiz_hideifs();
+
+        // Get existing form data.
+        $quizid = $quizform->get_coursemodule()->instance;
+        $existingsettings = $DB->get_record('quizaccess_seb_quizsettings', ['quizid' => $quizid], '*');
+
+        // Insert all the form elements before the 'security' section as a group.
+        foreach ($elements as $name => $type) {
```

Comment:
it feels like we can just $settingsmanager->get_quiz_element_types() here and remove $elements